File: question-5.js
const char = "*";
const space = " ";

const generateLine = (cursor, num) => {
  let line = "";
  const reflection = num + 1 - cursor;
  for (let i = 1; i <= num; i++) {
    switch (i) {
      case cursor:
      case reflection:
        line += char;
        break;
      default:
        line += space;
        break;
    }
  }
  return line;
};

const generateCross = (num) => {
  let solutions = [];
  for (let j = 1; j <= num; j++) {
    solutions.push(generateLine(j, num));
  }
  return solutions.join("\n");
};

console.log(generateCross(39));

File: question-5.go
package main

import ("fmt")

func generateCross(n int) {
  for i:=1; i<=n; i++ {
    for j:=1; j<=n; j++ {
      if i == j || n-i+1 ==j {
        fmt.Printf("*")
        continue
      }
      fmt.Printf(" ")
    }
    fmt.Println()
  }
}

func main() {
  generateCross(5)
}
